### PR TITLE
[SDPA] Create native function for determining which implementation of SDP to call

### PR DESCRIPTION
**Pull request by drisspg**
# Summary
Creates a callable native function that can determine which implementation of scaled dot product will get called. This allows to bump re-order the runtime dispatch of SDP to enable autograd.

### CODE REVIEW

**Comment by drisspg** [comment]
Still need to wait for the previously reverted failure to land then rebase. Was also going to utilize this new function but I think a follow up PR creates a cleaner break

**Review by cpuhrsch — APPROVED** [review]

File: torch/backends/cuda/__init__.py [hunk 1]
```diff
@@ -164,6 +165,19 @@ def preferred_linalg_library(backend: Union[None, str, torch._C._LinalgBackend]
...
+
+    .. warning:: This flag is experimental and subject to change.'
+
+    This class needs to stay inline with the enum defined in:
+    pytorch/aten/src/ATen/native/transformers/sdp_utils_cpp.h
+    """
+    MATH = 0
+    FLASH_ATTENTION = 1
+    EFFICIENT_ATTENTION = 2
+    ERROR = 3
```
Comment (cpuhrsch) [inline torch/backends/cuda/__init__.py:179]: Since this integer is signed, an error could also be "-1".

Comment (cpuhrsch) [reply torch/backends/cuda/__init__.py:179]: Then we don't have to do the following in the future

```
[...]
    EFFICIENT_ATTENTION = 2
    ERROR = 3
    NEW_ATTENTION_2 = 4
```

Comment (drisspg) [reply torch/backends/cuda/__init__.py]: great idea, just changed

**Pull request closed without merging**